### PR TITLE
Feat/sc 60 icons

### PR DESCRIPTION
**Pull request by davidgilbertson**
Jira: https://govausites.atlassian.net/browse/SC-60

Added an `<Icon>` component which made `<IndicatorTypeMark>` redundant.

The icons are defined as SVG. Things like the coat of arms and dta logo are not considered icons.

You can see the icons at `/style-guide`.

### CODE REVIEW

File: src/components/Card/CategoryCard/CategoryCard.scss [hunk 1]
```diff
@@ -8,7 +8,6 @@
 }
 
 .iconWrapper {
-  // TODO (davidg): implement real icons
```
Comment (davidgilbertson) [inline src/components/Card/CategoryCard/CategoryCard.scss:11]: This component isn't used anymore, it was in an earlier design. I'll delete the whole thing when the designs have stopped changing.

File: src/components/CategoryIndicator/CategoryIndicator.scss [hunk 2]
```diff
@@ -76,13 +70,3 @@
   font-size: 12px;
   font-weight: 700;
 }
-
-.viewIcon {
-  // TODO (davidg): icon and combine all icons
```
Comment (davidgilbertson) [inline src/components/CategoryIndicator/CategoryIndicator.scss]: In SC-16 you centralised these, but I've removed them all completely, so you can get rid of that mixin from SC-16 if you want.

File: src/components/IndicatorTypeMark/IndicatorTypeMark.test.js [hunk 3]
```diff
@@ -1,58 +0,0 @@
...
-      {...defaultProps}
-      size="LARGE"
-    />
-  );
-
-  const style = component.find('div').prop('style');
-  expect(style.background).toBeUndefined();
-  expect(style.border).toBe('1px solid JOBS_900');
-  expect(style.color).toBe('JOBS_900');
-});
```
Comment (davidgilbertson) [inline src/components/IndicatorTypeMark/IndicatorTypeMark.test.js:58]: So much deleted code. #happysad

File: src/components/StyleGuide/ColorRow/ColorRow.js [hunk 4]
```diff
@@ -7,7 +7,7 @@ const classnames = require('classnames/bind').bind(style);
 const ColorRow = props => (
   <div className={style.row}>
     {props.colors.map(color => (
-      <div className={style.swatchWrapper}>
+      <div key={color.variableName} className={style.swatchWrapper}>
```
Comment (davidgilbertson) [inline src/components/StyleGuide/ColorRow/ColorRow.js:10]: Unrelated. I don't know how, but I didn't have keys for any of these and never noticed the errors.

File: src/components/StyleGuide/StyleGuide.scss [hunk 5]
```diff
@@ -12,12 +12,36 @@
...
+.iconWrapper {
+  @include shadow;
+
+  display: inline-block;
+  padding: 16px 0;
+  margin: 0 16px 16px 0;
+  width: 200px;
+  text-align: center;
+  border-radius: 4px;
+  color: $color-primary--500;
```
Comment (davidgilbertson) [inline src/components/StyleGuide/StyleGuide.scss:36]: Colour the icons just so it's clear that the 'P' in the circle icon is hard-coded grey.

File: src/components/SubCategorySummary/SubCategorySummary.test.js [hunk 6]
```diff
@@ -3,12 +3,14 @@ import { shallow } from 'enzyme';
 import CategoryBanner from './SubCategorySummary';
 
 jest.mock('../Card/IndicatorCard/IndicatorCard', () => 'IndicatorCard');
+jest.mock('../Icon/Icon', () => 'Icon');
```
Comment (davidgilbertson) [inline src/components/SubCategorySummary/SubCategorySummary.test.js]: Mocking out the `<Icon>` in tests means that the snapshot of a component won't change if the icon component is changed.

File: src/constants/categories.js [hunk 7]
```diff
@@ -62,56 +69,33 @@ export default [
...
-        indicatorIds: [],
-      },
-      {
-        name: 'Life and death',
-        tint: '080',
-        shade: '800',
-        indicatorIds: [],
-      },
-      {
-        name: 'Socioeconomic disadvantage',
```
Comment (davidgilbertson) [inline src/constants/categories.js]: Deleting these, we only have two sub-categories per category now.

File: src/style/_misc.scss [hunk 8]
```diff
@@ -3,6 +3,16 @@ $animation-duration: 250ms;
...
+    box-shadow: 0 1px 3px rgba(0, 0, 0, 0.12), 0 1px 2px rgba(0, 0, 0, 0.24);
+  }
+
+  @if $elevation == 2 {
+    box-shadow: 0 3px 6px rgba(0, 0, 0, 0.16), 0 3px 6px rgba(0, 0, 0, 0.23);
+  }
+
+  @if $elevation == 3 {
+    box-shadow: 0 14px 28px rgba(0, 0, 0, 0.25), 0 10px 10px rgba(0, 0, 0, 0.22);
+  }
```
Comment (davidgilbertson) [inline src/style/_misc.scss:17]: New shadows, based on material design. Any existing use of `@include shadow` will default to the first one.

**Comment by micapam** [comment]
@davidgilbertson it looks like this branch is pretty broken. I can't run it locally, and circle is also complaining. Is there a missing file that needs to be checked in, or something ?

File: src/components/Icon/Icon.js [hunk 9]
```diff
@@ -0,0 +1,41 @@
...
+    style.svg,
+  );
+
+  return (
+    <svg
+      className={className}
+      width={`${props.size}px`}
+      height={`${props.size}px`}
+      viewBox={icon.viewBox}
+      style={props.style}
```
Comment (micapam) [inline src/components/Icon/Icon.js]: In all instances where `Icon` is used, I can only see `color` being set. Wouldn't it be more straightforward to simply have a `color` prop?

**Review by micapam — APPROVED** [review]
LGTM, one minor comment

File: src/components/Icon/Icon.js (hunk 9 above)
Comment (davidgilbertson) [reply src/components/Icon/Icon.js]: Good one, yeah I can't think what other style we'd ever need to pass in.

**Pull request merged**